### ISSUE TITLE
RuntimeError: maximum recursion depth exceeded while calling a Python object

### ISSUE THREAD

**Issue by SchubmannM** [MindscapeHQ/raygun4py#42]
Hi, I updated the newest version and tried using raygun as a middleware exception handler in my django project. 

Often I receive a RTE because of maximum recursion depth caused by the following method in raygunprovider.py:

def _post(self, raygunMessage):
        json = jsonpickle.encode(raygunMessage, unpicklable=False)

You can see the whole backtrace here: http://dpaste.com/3B3N98Q

**Comment by fundead** [comment]
Thanks for the report and the stack trace - we'll be adding more sophisticated guards to cater for this case. In the mean time would you be able to try disabling the variable checking by adding this to your init config block, to confirm if it is due to a circular reference in the variables:

```
'transmit_global_variables': False,
'transmit_local_variables': False
```

**Comment by SchubmannM** [comment]
Yes, turning the submission of global variables does fix it.

**Comment by fundead** [comment]
Great, thanks for confirming that. I have this noted down and will add a guard for this in the next release coming up.

**Comment by fundead** [comment]
Just an update on this one, a guard has been added for certain versions of jsonpickle failing to detect cycles in local/global variables upon serialization which should prevent this error. This has been released as v3.1.2. If anyone notices further occurrences of this please open another ticket.
